src/stream/src/common/log_store_impl/kv_log_store/writer.rs

```diff
@@ -165,16 +162,29 @@ impl<LS: LocalStateStore> LogWriter for KvLogStoreWriter<LS> {
         }
         flush_info.report(&self.metrics);
 
-        let truncate_offset = self.tx.pop_truncation(epoch);
+        if let Some(truncate_offset) = self.tx.pop_truncation(epoch) {
```

Comment:
Previously we only write vnode watermark to storage when we pop some new progress. After this PR, we will write a same table watermark to storage when there is no new progress, though the impact may be small.